### PR TITLE
Thret connect feed v3

### PR DESCRIPTION
**Pull request by gal-forer**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: Packs/FeedThreatConnect/Integrations/FeedThreatConnect/README.md [hunk 1]
```diff
@@ -1,106 +1,66 @@
...
-| feedReliability | Source Reliability | True |
-| tlp_color | The Traffic Light Protocol (TLP) designation to apply to indicators fetched from the feed. More information about the protocol can be found at https://us-cert.cisa.gov/tlp | False |
-| feedExpirationPolicy |  | False |
-| feedExpirationInterval |  | False |
-| feedFetchInterval | Feed Fetch Interval | False |
-| feedBypassExclusionList | Bypass exclusion list | False |
-| feedTags | Tags | False |
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | Base URL |  | True |
```
Comment (ShirleyDenkberg) [inline Packs/FeedThreatConnect/Integrations/FeedThreatConnect/README.md:12]: Need a description here.

File: Packs/FeedThreatConnect/Integrations/FeedThreatConnect/README.md [hunk 2]
```diff
@@ -1,106 +1,66 @@
...
+    | Access ID | The API ID key. | True |
+    | Secret key | The secrete key for the API. | True |
+    | Tags to filter results by | A comma-separated list of tags to filter by. | False |
+    | Owners | Comma-separated list of owners to fetch indicators from. | False |
+    | Fetch indicators |  | False |
+    | Indicator Reputation | Indicators from this integration instance will be marked with this reputation. | False |
+    | Source Reliability | Reliability of the source providing the intelligence data | True |
+    | Traffic Light Protocol Color | The Traffic Light Protocol \(TLP\) designation to apply to indicators fetched from the feed. | False |
+    |  |  | False |
+    |  |  | False |
```
Comment (ShirleyDenkberg) [inline Packs/FeedThreatConnect/Integrations/FeedThreatConnect/README.md:22]: These two lines need a description even if there is no parameter name.

**Comment by ShirleyDenkberg** [comment]
@gal-forer Need content for the pack README.

File: Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py [hunk 3]
```diff
@@ -174,15 +157,24 @@ def module_test_command(client: Client) -> COMMAND_OUTPUT:
         dict: Operation entry context - Empty.
         dict: Operation raw response - Empty.
     """
-    try:
-        client.get_indicators(limit=4)
-    except RuntimeError:
-        raise DemistoException("Unable to communicate with ThreatConnect API!")
+    url = '/api/v3/groups?resultLimit=4'
```
Comment (yaakovi) [inline Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py:160]: Do not pass query parameters as part of the URL; There is a dedicated argument for it.

File: Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py [hunk 4]
```diff
@@ -191,12 +183,56 @@ def fetch_indicators_command(client: Client) -> List[Dict[str, Any]]:
...
+        else:
+            return_error('Error from the API: ' + response.get('message',
+                                                               'An error has occurred if it persist please contact '
+                                                               'your local help desk'))
+
+    return [parse_indicator(indicator) for indicator in indicators]
 
-    return [parse_indicator(indicator) for indicator in raw_response]
 
+def create_or_query(delimiter_str: str, param_name: str) -> str:
```
Comment (yaakovi) [inline Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py:225]: Move this function above the `fetch_groups_command` function
[Function and parameter ordering](https://docs.google.com/document/d/1PTmRCb_aksrMreJrRsVr90r_4wTHTQaMJKP3P6aS6Ks/edit#heading=h.sx0rbaugwwjw)
```suggestion
def create_or_query(param_name: str, delimited_str: str) -> str:
```

File: Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py [hunk 5]
```diff
@@ -191,12 +183,56 @@ def fetch_indicators_command(client: Client) -> List[Dict[str, Any]]:
...
+          f'{group_type if group_type != "AND () " else ""}{status if status != "AND () " else ""}' \
+          f'{from_date if from_date != "AND () " else ""}'.replace('AND', '', 1)
+    if tql:
+        tql = urllib.parse.quote(tql.encode('utf8'))  # type: ignore
+        tql = f'?tql={tql}'
+    else:
+        tql = ''
+        if fields:
+            fields = fields.replace('&', '?', 1)  # type: ignore
+    url = f'/api/v3/groups{tql}{fields}&resultStart={page}&resultLimit=500'
```
Comment (yaakovi) [inline Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py:207]: Do not pass query parameters as part of the URL; There is a dedicated argument for it.

File: Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py [hunk 6]
```diff
@@ -191,12 +183,56 @@ def fetch_indicators_command(client: Client) -> List[Dict[str, Any]]:
...
+    group_type = f'AND ({create_or_query(demisto.params().get("group_type", "Incident"), "typeName")}) '
+    first_fetch_time = demisto.getParam("first_fetch_time")
+    from_date = ''
+    if first_fetch_time:
+        first_fetch_time = dateparser.parse(demisto.getParam('first_fetch_time').strip()).strftime("%Y-%m-%d")  # type: ignore # noqa
+        from_date = f'AND (dateAdded > "{first_fetch_time}") '
+    page = 0
+    tql = f'{owners if owners != "AND () " else ""}{tags if tags != "AND () " else ""}' \
+          f'{group_type if group_type != "AND () " else ""}{status if status != "AND () " else ""}' \
+          f'{from_date if from_date != "AND () " else ""}'.replace('AND', '', 1)
```
Comment (yaakovi) [inline Packs/FeedThreatConnect/Integrations/FeedThreatConnect/FeedThreatConnect.py:199]:
```suggestion
    tql = (
        f'{owners if owners != "AND () " else ""}'
        f'{tags if tags != "AND () " else ""}'
        f'{group_type if group_type != "AND () " else ""}'
        f'{status if status != "AND () " else ""}'
        f'{from_date if from_date != "AND () " else ""}'
    ).replace('AND', '', 1)  # replace the preceding 'AND'
          
```

**Comment by ShirleyDenkberg** [comment]
@yaakovi I have finished my review of the docs in this PR.  But, @gal-forer we still need content for the content pack readme.

File: Packs/FeedThreatConnect/ReleaseNotes/2_0_0.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### ThreatConnect Feed
+- Added support for the new v3 API.
+- Updated the Docker image to: *demisto/threatconnect-tcex:1.0.0.26622*.
```
[the remark this answers is not in the document]
Comment (gal-forer) [reply Packs/FeedThreatConnect/ReleaseNotes/2_0_0.md:5]: No, I've changed it to the python3 one

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

**Review by yaakovi — APPROVED** [review]

**Pull request merged**